rocky-raft/src/rocky/raft/client/RaftClient.java

```diff
@@ -64,10 +65,16 @@ private ServerConfig findLeader() throws Exception {
 
     @Override
     public void post(String message) throws Exception {
+        post(message, Utils.getRandomLong());
+    }
+
+    @Override
+    public void post(String message, long id) throws Exception {
```

Comment:
And we can assume that id is always supplied. SuperClient can generate UUID and pass it here.